.pre-commit-config.yaml

```diff
@@ -37,4 +51,106 @@ repos:
         files: ^frontend/src/.*(.js|\.jsx|\.ts|\.tsx)$
         language: node
         pass_filenames: true
-
+# TODO: Uncomment when all files have correct license headers
+#      - id: license-headers
+#        name: Checks that all files have the required license headers
+#        entry: ./scripts/check_license_headers.py
+#        language: system
+#        always_run: true
+#        pass_filenames: false
+  - repo: https://github.com/Lucas-C/pre-commit-hooks
+    rev: v1.2.0
+    hooks:
+      - id: insert-license
+        name: Add license for all JS(X)/TS(X) files
+        files: \.(s?css|jsx?|tsx?)$
+        args:
+          - --comment-style
+          - "/**| *| */"
+          - --license-filepath
+          - scripts/license-template.txt
+          - --fuzzy-match-generates-todo
+        exclude: |
+          (?x)
+          /vendor/|
+          ^component-lib/declarations/apache-arrow
+      - id: insert-license
+        name: Add license for all Proto files
+        files: \.proto$
+        args:
+          - --comment-style
+          - "/**!| *| */"
+          - --license-filepath
+          - scripts/license-template.txt
+          - --fuzzy-match-generates-todo
+        exclude: |
+          (?x)
+          /vendor/|
+          ^component-lib/declarations/apache-arrow
```

Comment:
Is there any way to prevent duplicating this exclusion?

Reply:
I'm afraid there is no easy way to do this.